chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/CacheInitializorImpl.java

```diff
@@ -0,0 +1,49 @@
+package gov.healthit.chpl.domain;
+
+import java.io.IOException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.test.context.ContextConfiguration;
+
+import gov.healthit.chpl.auth.permission.GrantedPermission;
+import gov.healthit.chpl.auth.user.JWTAuthenticatedUser;
+import gov.healthit.chpl.dao.EntityRetrievalException;
+import gov.healthit.chpl.web.controller.CertificationIdController;
+import gov.healthit.chpl.web.controller.SearchViewController;
+
+@ContextConfiguration(classes = { gov.healthit.chpl.CHPLConfig.class })
+@Component
+public class CacheInitializorImpl {
+	private static JWTAuthenticatedUser adminUser;
+	private static final Logger logger = LogManager.getLogger(CacheInitializorImpl.class);
+	
+	@Autowired private CertificationIdController certificationIdController;
+	@Autowired private SearchViewController searchViewController;
```

Comment:
Accessing controllers this way really violates Spring's MVC principles. Check out the discussion here http://stackoverflow.com/questions/20374972/invoke-controller-method-from-java-class. I would really try to use the managers instead - if some business logic needs to be moved from the controllers into the managers so that can be done, that's fine.